packages/ckeditor5-ui/src/focuscycler.ts

```diff
@@ -327,7 +327,7 @@ export default class FocusCycler extends EmitterMixin() {
 			index = ( index + collectionLength + step ) % collectionLength;
 		} while ( index !== current );
 
-		return null;
+		return this.focusables.get( this.current! ) as FocusableView;
```

Comment:
We had some F2F discussion about this loop, whether it can go infinite, etc. We also did some changes to types in other PR and we discussed that `focusables` should indeed always have only `FocusableView`s.

I wanted to drop `index !== current` condition from the loop so that `current` is also considered so this return will not need to change.

However, the loop can go infinite if all focusables are hidden.

But in this case, if all focusables are hidden (including current) should we return current or `null`?

We could change this to return `null` if current is not focusable but I am not 100% sure if that's correct -- I am not sure if that would work correctly with some chained focus cyclers etc. So I leave this thought to you, please proceed as you like.

Reply:
I feel an instant aversion to unrestricted loops 🙂 So instead I went with 

```ts
let index = ( current + collectionLength + step ) % collectionLength;

for ( let i = 0; i++ < collectionLength ) {
	const view = this.focusables.get( index )!;

	if ( isFocusable( focusableItemCandidate ) ) {
		return view;
	}

	index = ( index + collectionLength + step ) % collectionLength;
}

// TS needs this...
return null
```

which was OK if not for the fact that TS does not understand that in this loop we'll always get back to the current (initial) view that is focusable. I had to add `null` below just to satisfy TS it and I think it's misleading for anyone who'll read this code in the future.

Finally I went with the `do {} while ()` implementation but with slight refactoring and some comments [780a93e](https://github.com/ckeditor/ckeditor5/pull/15623/commits/780a93e1ba4bc75eddbe290d8c35e6bb66211445).